modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/publisher/APIM574ChangeTheStatusOfAnAPIToPrototypedThroughThePublisherRestAPITestCase.java

```diff
@@ -109,132 +100,84 @@ public void testCreateAnAPIThroughThePublisherRest() throws Exception {
         apiCreationRequestBean.setBizOwnerMail("api574b@ee.com");
         apiCreationRequestBean.setTechOwner("api574t");
         apiCreationRequestBean.setTechOwnerMail("api574t@ww.com");
-
-        HttpResponse apiCreationResponse = apiPublisher.addAPI(apiCreationRequestBean);
-        JSONObject apiResponse = new JSONObject(apiCreationResponse.getData());
-        assertEquals(apiCreationResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
-                "Response Code miss matched when creating the API");
-        assertFalse(apiResponse.getBoolean("error"), apiNameTest + "is not created as expected");
-
+        APIDTO apiCreationResponse = restAPIPublisher.addAPI(apiCreationRequestBean);
+        String status = apiCreationResponse.getLifeCycleStatus();
+        apiId = apiCreationResponse.getId();
         //Check availability of the API in publisher
-        HttpResponse apiResponsePublisher = apiPublisher.getAPI(apiNameTest, apiProvider, apiVersion);
-        JSONObject jsonObject = new JSONObject(apiResponsePublisher.getData());
-        JSONObject apiObject = new JSONObject(jsonObject.getString("api"));
-        assertFalse(jsonObject.getBoolean("error"), apiNameTest + " is not visible in publisher");
-        assertTrue(apiObject.getString("name").equals(apiNameTest),
-                apiNameTest + " is not visible in publisher");
-        assertTrue(apiObject.getString("status").equals("CREATED"),
-                "Status of the " + apiNameTest + "is not a valid status");
-
+        HttpResponse apiResponsePublisher = restAPIPublisher.getAPI(apiId);
+        assertEquals(apiResponsePublisher.getResponseCode(), Response.Status.OK.getStatusCode(), apiNameTest +
+                " is not visible in publisher");
+        assertTrue(apiCreationResponse.getName().equals(apiNameTest), apiNameTest + " is not visible in " +
+                "publisher");
+        assertTrue(status.equals("CREATED"), "Status of the " + apiNameTest + "is not a valid status");
```

Comment:
Lets use the lifecycle status constant here

Reply:
fixed